src/libraries/Microsoft.Extensions.Caching.Memory/tests/MemoryCacheSetAndRemoveTests.cs

```diff
@@ -793,6 +793,40 @@ public async Task GetOrCreateAsyncFromCacheWithNullKeyThrows()
             var cache = CreateCache();
             await Assert.ThrowsAsync<ArgumentNullException>(async () => await cache.GetOrCreateAsync<object>(null, null));
         }
+        
+        [Fact]
+        public void GetOrCreateWithCacheEntryOptions()
+        {
+            var cacheKey = "test";
+            var cache = CreateCache();
+            var expiry = 1000;
+            var cacheEntryOptions = new MemoryCacheEntryOptions()
+            {
+                AbsoluteExpirationRelativeToNow = TimeSpan.FromMilliseconds(expiry)
+            };            
+            var value = cache.GetOrCreate<string>(cacheKey, _ => cacheKey, cacheEntryOptions);
+            Assert.Equal(cacheKey, value);
+            Assert.True(cache.TryGetValue(cacheKey, out _));
+            Thread.Sleep(expiry * 2);
```

Comment:
I would rather not add more timeout tests because they tend to fail on slow machines. Plus a two seconds way is a bit large for CI, is there a way we can test without expiration?

Reply:
```cs
[Fact]
public async Task GetOrCreateAsyncWithCacheEntryOptions()
{
    var cacheKey = "test";
    var cache = CreateCache();
    ManualResetEvent mre = new ManualResetEvent(false);
    
    var options = new MemoryCacheEntryOptions();
    options.PostEvictionCallbacks.Add(new PostEvictionCallbackRegistration()
    {
        EvictionCallback = (key, value, reason, state) =>
        {
            Assert.Equal(cacheKey, key);
            Assert.Equal(cacheKey, value);
            Assert.Equal(EvictionReason.Removed, reason);
            mre.Set();
        }
    });

    var value = await cache.GetOrCreateAsync<string>(cacheKey, _ => Task.FromResult(cacheKey), cacheEntryOptions);
    Assert.Equal(cacheKey, value);
    Assert.True(cache.TryGetValue(cacheKey, out _));
    
    cache.Remove(cacheKey);
    Assert.True(mre.WaitOne(TimeSpan.FromSeconds(30)));
    Assert.False(cache.TryGetValue(cacheKey, out _));
}
```